packages/cli/src/commands/promote.ts

```diff
@@ -0,0 +1,123 @@
+import chalk from 'chalk';
+import type Client from '../util/client';
+import getArgs from '../util/get-args';
+import getProjectByCwdOrLink from '../util/projects/get-project-by-cwd-or-link';
+import { getPkgName } from '../util/pkg-name';
+import handleError from '../util/handle-error';
+import { isErrnoException } from '@vercel/error-utils';
+import logo from '../util/output/logo';
+import ms from 'ms';
+import requestPromote from '../util/promote/request-promote';
+import promoteStatus from '../util/promote/status';
+
+const help = () => {
+  console.log(`
+  ${chalk.bold(`${logo} ${getPkgName()} promote`)} [deploymentId|deploymentName]
+
+  Promote an existing deployment to current.
+
+  ${chalk.dim('Options:')}
+
+    -h, --help                     Output usage information
+    -A ${chalk.bold.underline('FILE')}, --local-config=${chalk.bold.underline(
+    'FILE'
+  )}   Path to the local ${'`vercel.json`'} file
+    -Q ${chalk.bold.underline('DIR')}, --global-config=${chalk.bold.underline(
+    'DIR'
+  )}    Path to the global ${'`.vercel`'} directory
+    -d, --debug                    Debug mode [off]
+    --no-color                     No color mode [off]
+    -t ${chalk.bold.underline('TOKEN')}, --token=${chalk.bold.underline(
+    'TOKEN'
+  )}        Login token
+    --timeout=${chalk.bold.underline(
+      'TIME'
+    )}                 Time to wait for promotion completion [3m]
+    -y, --yes                      Skip questions when setting up new project using default scope and settings
+
+  ${chalk.dim('Examples:')}
+
+  ${chalk.gray('–')} Show the status of any current pending promotions
+
+    ${chalk.cyan(`$ ${getPkgName()} promote`)}
+    ${chalk.cyan(`$ ${getPkgName()} promote status`)}
+    ${chalk.cyan(`$ ${getPkgName()} promote status <project>`)}
+    ${chalk.cyan(`$ ${getPkgName()} promote status --timeout 30s`)}
+
+  ${chalk.gray('–')} Promote a deployment using id or url
+
+    ${chalk.cyan(`$ ${getPkgName()} promote <deployment id/url>`)}
+`);
+};
+
+/**
+ * `vc promote` command
+ * @param {Client} client
+ * @returns {Promise<number>} Resolves an exit code; 0 on success
+ */
+export default async (client: Client): Promise<number> => {
+  let argv;
+  try {
+    argv = getArgs(client.argv.slice(2), {
+      '--debug': Boolean,
+      '-d': '--debug',
```

Comment:
<!-- Suggestion comments propose improvements to the code. Mention specifically what should be changed and what the benefit would be.  -->
**suggestion (soft-blocking):** remove common args

In this or a follow-up PR: The `--debug` option is a common arg. It shouldn't be necessary to list it here. Can we remove it?
‏
```suggestion
```